src/lib.rs

```diff
@@ -706,18 +668,107 @@ where
     Data: Serialize + DeserializeOwned + Clone + Send,
     DeSer: DeSerializer<Data> + Send + Sync + Clone,
 {
+    /// Create new [`FileDatabase`] from the file at [`Path`](std::path::Path),
+    /// and load the contents.
+    pub fn load_from_path<S>(path: S) -> error::Result<Self>
+    where
+        S: AsRef<std::path::Path>,
+    {
+        let mut backend = FileBackend::from_path_or_fail(path)?;
+        let deser = DeSer::default();
+        let data = Self::load_from_backend(&mut backend, &deser)?;
+
+        let db = Self {
+            data: RwLock::new(data),
+            backend: Mutex::new(backend),
+            deser,
+        };
+        Ok(db)
+    }
+
+    /// Load [`FileDatabase`] at `path` or initialise with `data`.
+    ///
+    /// Create new [`FileDatabase`] from the file at [`Path`](std::path::Path),
+    /// and load the contents. If the file does not exist, initialise with
+    /// `data`.
+    pub fn load_from_path_or<S>(path: S, data: Data) -> error::Result<Self>
+    where
+        S: AsRef<std::path::Path>,
+    {
+        let (mut backend, exists) = FileBackend::from_path_or_create(path)?;
+        let deser = DeSer::default();
+        if !exists {
+            let ser = deser.serialize(&data).context(ErrorKind::Serialization)?;
+            backend.put_data(&ser)?;
+        }
+
+        let db = Self {
+            data: RwLock::new(data),
+            backend: Mutex::new(backend),
+            deser,
+        };
+
+        if exists {
+            db.load()?;
+        }
+
+        Ok(db)
+    }
+
+    /// Load [`FileDatabase`] at `path` or initialise with `closure`.
+    ///
+    /// Create new [`FileDatabase`] from the file at [`Path`](std::path::Path),
+    /// and load the contents. If the file does not exist, `closure` is
+    /// called and the database is initialised with it's return value.
+    pub fn load_from_path_or_else<S, C>(path: S, closure: C) -> error::Result<Self>
+    where
+        S: AsRef<std::path::Path>,
+        C: FnOnce() -> Data,
```

Comment:
What do you think of having an `Option<Vec<u8>>` as argument to the closure? This would potentially allow users to upgrade old data schemes.

Reply:
What should be passed to the closure? Note that we only call `closure` if the file didn't yet exist.
I think most use cases are already covered by `Database::convert`. Always calling the closure would not be in line with the name (**`or_else`**).
If someone needs something *very specific* it shouldn't be hard for themselves to do it directly with the backend (although it would then be useful to make `Database::load_from_backend` public API).

Reply:
Fair enough. Let's keep it as it is.